compute_endpoint/tests/unit/test_reporting_period.py

```diff
@@ -30,13 +25,24 @@ def test_default_period():
     assert gce.executor.heartbeat_period == 1
 
 
-def test_reporting_period():
-    thread = ReportingThread(target=counter, args=[])
-
-    with patch.object(thread, "reporting_period", REPORTING_PERIOD):
-        assert thread.reporting_period == REPORTING_PERIOD
+@pytest.mark.parametrize("reporting_period", (None, 0.1, 0.2, 1, 5, 11, 30.0))
+def test_reporting_period(reporting_period):
```

Comment:
test_reporting_period is kinda generic.  you changed the logic by quite a bit, is there a more specific test name?

Reply:
I decided to leave it.  The original intent of the test was, I think, to verify that the reporting period was called periodically.  The update here is to trust the external API does what it says it does, and verify our part via mocking: do we _call the API correctly_?

I've similarly removed the `None` from this one, as that's what the first test is verifying: the default value.